ci/jobs/satellite6-pulp.yaml

```diff
@@ -0,0 +1,187 @@
+            echo "INSTANCE_IP=${{INSTANCE_IP}}" >> openstack.properties
+            echo "INSTANCE_ID=${{INSTANCE_ID}}" >> openstack.properties
+            echo "SECURITY_GROUP_NAME=${{SECURITY_GROUP_NAME}}" >> openstack.properties
+            echo "REMOTE_JENKINS_USERNAME=${{REMOTE_JENKINS_USERNAME}}" \
+                > remote-jenkins.properties
+            echo "REMOTE_JENKINS_API_TOKEN=${{REMOTE_JENKINS_API_TOKEN}}" \
+                >> remote-jenkins.properties
+            echo "REMOTE_JENKINS_URL=${{REMOTE_JENKINS_URL}}" \
+                >> remote-jenkins.properties
+        - inject:
+            properties-file: openstack.properties
+        - inject:
+            properties-file: remote-jenkins.properties
+        - shell: |
+            # Wait until the OpenStack instance can be connected by SSH sessions.
+            # The following trigger builders will depend on SSH connectivity.
+            set +e
+            while true; do
+                ssh -o ConnectTimeout=5 \
+                    -o StrictHostKeyChecking=no \
+                    cloud-user@${{INSTANCE_IP}} echo testing
+                if [[ $? = 0 ]]; then
+                    break
+                fi
+                echo "$INSTANCE_IP is not yet ready...";
+                sleep 5
+            done
+            set -e
+        - trigger-builds:
+            - project: "satellite6-pulp-trigger"
+              block: true
+              bool-parameters:
+                  - name:  SETUP_FAKE_MANIFEST_CERTIFICATE
+                    value: true
+              predefined-parameters: |
+                  TRIGGER_TYPE=installation
+                  SERVER_HOSTNAME=${{INSTANCE_IP}}
+                  DISTRIBUTION=DOWNSTREAM
+                  SATELLITE_VERSION=6.2
+                  SATELLITE_RELEASE=GA
+                  REMOTE_JENKINS_USERNAME=${{REMOTE_JENKINS_USERNAME}}
+                  REMOTE_JENKINS_API_TOKEN=${{REMOTE_JENKINS_API_TOKEN}}
+                  REMOTE_JENKINS_URL=${{REMOTE_JENKINS_URL}}
+        - trigger-builds:
+            - project: "satellite6-pulp-upgrade"
+              block: true
+              predefined-parameters: |
+                  INSTANCE_IP=${{INSTANCE_IP}}
+        - trigger-builds:
+            - project: "satellite6-pulp-trigger"
+              block: true
+              predefined-parameters: |
+                  TRIGGER_TYPE=robottelo
+                  SERVER_HOSTNAME=${{INSTANCE_IP}}
+                  TEST_TYPE=endtoend-api
+                  REMOTE_JENKINS_USERNAME=${{REMOTE_JENKINS_USERNAME}}
+                  REMOTE_JENKINS_API_TOKEN=${{REMOTE_JENKINS_API_TOKEN}}
+                  REMOTE_JENKINS_URL=${{REMOTE_JENKINS_URL}}
+        - shell: |
+            echo done
+            echo ${{REMOTE_JENKINS_USERNAME}}
+            source sat-pulp/bin/activate
+            python -c "import sys; print hasattr(sys, 'real_prefix')"
+            source $OPENSTACK_CREDENTIALS
+            nova list | grep ${{INSTANCE_ID}}
```

Comment:
You should use an if block here. If the instance can't be found than this will make the job fail.